test/test_utilities.cpp

```diff
@@ -191,16 +191,24 @@ void Compare(Tensor& T1, Tensor& T2, float tol) {
   for (int k = 0; k < T_size; k++) {
     auto a = T1_data[k];
     auto b = T2_data[k];
-    if (!(std::isnan(a) && std::isnan(b)) &&
-        !(std::isinf(a) && std::isinf(b))) {
-      if (a == 0) {
-        EXPECT_NEAR(a, b, tol);
+
+    if (std::isnan(a) && std::isnan(b)) {
+      continue;
+    }
+    if (std::isinf(a) && std::isinf(b)) {
+      if ((a < 0 && b < 0) || (a > 0 && b > 0)) {
+        continue;
       } else {
-        auto rel = a - b;
-        auto rel_div = std::abs(rel / a);
-        EXPECT_TRUE(rel_div <= tol);
+        ASSERT_TRUE(false) << "Mismatch inf and -inf";
```

Comment:
Minor:
We can replace the if-else with:
`EXPECT_TRUE((a < 0 && b < 0) || (a > 0 && b > 0)) << "Mismatch inf and -inf";`